apidef/migration_test.go

```diff
@@ -315,3 +315,79 @@ func TestAPIDefinition_MigrateEndpointMeta(t *testing.T) {
 	assert.Equal(t, expectedWhitelist, api.VersionData.Versions[""].ExtendedPaths.Ignored)
 	assert.Equal(t, expectedMockResponse, api.VersionData.Versions[""].ExtendedPaths.MockResponse)
 }
+
+func TestAPIDefinition_MigrateCache(t *testing.T) {
+	name := ""
+
+	versionInfo := VersionInfo{
+		Name:             name,
+		UseExtendedPaths: true,
+		ExtendedPaths: ExtendedPathsSet{
+			Cached: []string{"test"},
+		},
+	}
+
+	old := func() APIDefinition {
+		return APIDefinition{
+			VersionDefinition: VersionDefinition{
+				Location: URLLocation,
+			},
+			VersionData: VersionData{
+				NotVersioned:   false,
+				DefaultVersion: name,
+				Versions: map[string]VersionInfo{
+					name: versionInfo,
+				},
+			},
+		}
+	}
+
+	check := func(t *testing.T, base APIDefinition) {
+		err := base.MigrateCachePlugin()
+		assert.NoError(t, err)
+
+		defaultVersionInfo := base.VersionData.Versions[base.VersionData.DefaultVersion]
+		path1 := defaultVersionInfo.ExtendedPaths.AdvanceCacheConfig[0]
+		path1Cache := defaultVersionInfo.ExtendedPaths.Cached[0]
+
+		expectedCachedDefault := []string{"test"}
+		cacheItemGet := CacheMeta{
+			Method:        http.MethodGet,
+			Disabled:      false,
+			Path:          "test",
+			CacheKeyRegex: "",
+		}
+		cacheItemHead := CacheMeta{
+
+			Disabled:      false,
+			Path:          "test",
+			CacheKeyRegex: "",
+			Method:        http.MethodHead,
+		}
+		cacheItemOptions := CacheMeta{
+			Disabled:      false,
+			Path:          "test",
+			CacheKeyRegex: "",
+			Method:        http.MethodOptions,
+		}
+		expectedAdvCacheMethods := []CacheMeta{
+			cacheItemGet,
+			cacheItemHead,
+			cacheItemOptions,
+		}
+
+		// for every cache item we have to have 3 new AdvanceCacheConfig items
+		// since we do support caching only for safe methods GET, OPTIONS and HEAD
+		assert.Equal(t, len(defaultVersionInfo.ExtendedPaths.AdvanceCacheConfig), len(defaultVersionInfo.ExtendedPaths.Cached)*3)
+		assert.Equal(t, path1.Path, path1Cache)
+
+		assert.Equal(t, expectedCachedDefault, defaultVersionInfo.ExtendedPaths.Cached)
+		assert.Equal(t, expectedAdvCacheMethods, defaultVersionInfo.ExtendedPaths.AdvanceCacheConfig)
+	}
+
+	t.Run("check migration of old data to new", func(t *testing.T) {
```

Comment:
why did we put it inside a sub-test? This is an unnecessary layer. If we'd test more than one thing we'd add but this layer is the main one.